### PR TITLE
Add QSPI support for the SAMD51

### PR DESCRIPTION
**Pull request by tannewt**
This enables faster flash reading and writing. The latest revs of the Metro M4 use QSPI to enable faster interactions with the flash.

### CODE REVIEW

File: ports/atmel-samd/external_flash/external_flash.c [hunk 1]
```diff
@@ -300,22 +195,20 @@ void spi_flash_init(void) {
...
+         || jedec_id_response[0] == SPI_FLASH_JEDEC_MANUFACTURER_2
 #endif
          ) &&
-        jedec_id_response[2] == SPI_FLASH_JEDEC_MEMORY_TYPE &&
-        jedec_id_response[3] == SPI_FLASH_JEDEC_CAPACITY) {
+        jedec_id_response[1] == SPI_FLASH_JEDEC_MEMORY_TYPE &&
+        jedec_id_response[2] == SPI_FLASH_JEDEC_CAPACITY) {
         spi_flash_is_initialised = true;
     } else {
+        asm("bkpt");
```
Comment (dhalbert) [inline ports/atmel-samd/external_flash/external_flash.c]: Is this a leftover debugging statement?

**Comment by dhalbert** [comment]
Does metro_m4_express now work with the old Metro boards that we and the helpers have? If not maybe we should make up another board type at least temporarily.

**Comment by dhalbert** [comment]
Last commit says "m4 qspi works. m0 compiles" and is from 3 days ago. Is that the last one?

File: ports/atmel-samd/external_flash/external_flash.c (hunk 1 above)
Comment (tannewt) [reply ports/atmel-samd/external_flash/external_flash.c]: Yup!

**Comment by tannewt** [comment]
This will no longer work with the black Rev B out of the box. Its not too hard to switch back though.

That is the latest commit. I took a break over the weekend and tested the metro m0 express today before sending the PR. It worked just fine so I didn't need another commit.

Thanks for the review!

**Comment by dhalbert** [comment]
I was trying to make a `metro_m4_express_revb` board temporary board def. I just copied the new version and changed QSPI_FLASH_FILESYSTEM` to `SPI_FLASH_FILESYSTEM` in `mpconfigboard.mk`.  (I noticed you corrected the `MICROPY_PORT_x` defns which describe which pins not to reset, and which appear to have been wrong for a while.)

However, what happens now is that the M4 RevB gets a blue (?color vision issues) status LED every hard-reset and always re-creates the filesystem (takes a number of seconds to do this -- be patient with the blue LED changing to pulsing green).

This problem does not occur on the M0.

**Pull request merged**